features/upgrade/api_auth/upgrade.feature

```diff
@@ -173,3 +173,54 @@ Feature: apiserver and auth related upgrade check
       | o             | yaml                        |
     Then the expression should be true> @result[:parsed]['subjects'][0]['name'] == "system:authenticated"
     And the expression should be true> @result[:parsed]['subjects'][1]['name'] == "system:unauthenticated"
+
+  # @author scheng@redhat.com
+  # @case_id OCP-29741
+  @upgrade-prepare
+  @admin
+  Scenario: After cluster upgrade,the changes to the default SCCs should not be stomped by CVO - prepare
+    Given as admin I successfully merge patch resource "scc/anyuid" with:
+      | {"users": ["system:serviceaccount:test-scc:test-scc"]} |
+    Given as admin I successfully merge patch resource "scc/privileged" with:
+      | {"users": ["system:admin","system:serviceaccount:openshift-infra:build-controller","system:serviceaccount:test-scc:test-scc"]} |
```

Comment:
IMO, changing scc is a destructive operation to other scenarios. @xingxingxia  Could you confirm ?

Reply:
Missing SCC upgrade case brought serious customer issue such that we were asked to do post morterm and retrospective in google doc https://url.corp.redhat.com/33223e4 . Now with issue fixed, not destructive as long as no new regression bug happens. Thus needs the case.

Reply:
@liangxia , @barleyer , I didn't look deeper at https://github.com/openshift/verification-tests/pull/1128#pullrequestreview-425869448 . Now after more looking, seems original version is safer and preferred than using `as admin I successfully merge patch resource "scc/privileged" with:` because original version adds an item to the array, no matter whether the array's default items change or not in future OCP.

Reply:
Thus either use original version, or core auto guys help implement "as admin I successfully **json** patch resource "scc/privileged" with:"

Reply:
I'm not talking we should add destructive test cases or not for upgrade.
I mean, if the test case is destructive, then we should tag it destructive. Generally, we run destructive test cases after non-destructive ones, so there are less interfere.

By the way, "as admin I successfully json patch resource" is already there, the patch_type isn't hard-coded.
```
Given /^(?:(as admin) )?I successfully#{OPT_WORD} patch resource "(.*)\/(.*)" with:$/ do |as_admin, patch_type, resource_type, resource_name, table|
```

Reply:
OK, thank you for clarification. I didn't notice that yet.

Reply:
@liangxia tag added.

Reply:
I have no idea the diff of json patch and merge patch.
I'm merging now. You can open another PR for that if you figure out another one is more appropriate.

Reply:
Hmm, too quickly. Then OK let another PR update the lines with **json** patch

Reply:
@barleyer could you update with another PR?

Reply:
Hey,I tried many time on type `json`,It all failed.This is a destructive test case,I think It wont Influence others.

Reply:
For type `json`,I think the below codes cannt judge if the patch action if execute right.
```
236   success = wait_for(sec) {
237     failpath = []
238     res = _user.cli_exec(:get, **opts)
239     substruct?(expect_hash, res[:parsed], vague_nulls: true, failpath: failpath, exact_arrays: true, null_deletes_key: true)
240   }
```

Reply:
The `expect_hash` will contains some parameter like `'[{"op": "add", "path": "/users/-", "value": "system:serviceaccount:test-scc:test-scc"}]'` and this won't appear in `res = _user.cli_exec(:get, **opts)`